acceptance/lib/facter/acceptance/base_fact_utils.rb

```diff
@@ -0,0 +1,519 @@
+  module Acceptance
+    module BaseFactUtils
+
+      # return a hash of expected facts for os, processors, and kernel facts
+      #
+      def os_processors_and_kernel_expected_facts(agent)
+        if agent['platform'] =~ /aix-/
+          aix_expected_facts(agent)
+        elsif agent['platform'] =~ /cisco/
+          cisco_expected_facts(agent)
+        elsif agent['platform'] =~ /cumulus-/
+          cumulus_expected_facts(agent)
+        elsif agent['platform'] =~ /debian-/
+          debian_expected_facts(agent)
+        elsif agent['platform'] =~ /el-|centos-/
+          el_expected_facts(agent)
+        elsif agent['platform'] =~ /eos-/ # arista
+          eos_expected_facts(agent)
+        elsif agent['platform'] =~ /fedora-/
+          fedora_expected_facts(agent)
+        elsif agent['platform'] =~ /huaweios-/
+          huawei_expected_facts(agent)
+        elsif agent['platform'] =~ /osx-/
+          osx_expected_facts(agent)
+        elsif agent['platform'] =~ /sles-/
+          sles_expected_facts(agent)
+        elsif agent['platform'] =~ /solaris-/
+          solaris_expected_facts(agent)
+        elsif agent['platform'] =~ /ubuntu-/
+          ubuntu_expected_facts(agent)
+        elsif agent['platform'] =~ /windows-/
+          windows_expected_facts(agent)
+        else
+          fail_test("unknown agent type being tested #{agent['platform']}")
+        end
+      end
+
+      # return the value from the facter json parsed set of results using the fact path separated using '.'s
+      #
+      def json_result_fact_by_key_path(results, fact)
+        fact.split('.').each do |key|
+          results = results[key]
+        end
+        results
+      end
+
+      # AIX
+      def aix_expected_facts(agent)
+        version = agent['platform'].match(/aix-(\d)\.(\d)/)
+        kernel_major_version = "#{version[1]}#{version[2]}00"
+        kernel_release = /^#{kernel_major_version}-\d+-\d+-\d+/
+        os_arch     = /[Pp]ower[Pp][Cc]/
+        os_hardware = /IBM/
+
+        expected_facts = {
+            'os.architecture'   => os_arch,
+            'os.family'         => 'AIX',
+            'os.hardware'       => os_hardware,
+            'os.name'           => 'AIX',
+            'os.release.full'   => kernel_release,
+            'os.release.major'  => kernel_major_version,
+            'processors.count'  => /[1-9]+/,
+            'processors.isa'    => os_arch,
+            'processors.models' => os_arch,
+            'kernel'            => 'AIX',
+            'kernelrelease'     => kernel_release,
+            'kernelversion'     => kernel_major_version,
+            'kernelmajversion'  => kernel_major_version
+        }
+        expected_facts
+      end
+
+      # Cisco
+      def cisco_expected_facts(agent)
+        case agent['platform']
+          when /cisco_ios/
+            os_version         = '6'
+            os_name            = 'ios_xr'
+          when /cisco_nexus/
+            os_version         = '7'
```

Comment:
@LuvCurves are these the only supported cisco versions or just the ones we have access to. In other words, will constraining this specifically cause Cisco's internal Beaker runs to go :boom: .